runtime/docker/docker.go

```diff
@@ -71,10 +72,17 @@ func New(opts ...ClientOpt) (*client, error) {
 		}
 	}
 
+	// create a new retryable http client
+	//
+	// https://pkg.go.dev/github.com/hashicorp/go-retryablehttp#NewClient
+	retryClient := retryablehttp.NewClient()
+	// set logger to nil to avoid spam
+	retryClient.Logger = nil
+
 	// create new Docker client from environment
 	//
 	// https://godoc.org/github.com/docker/docker/client#NewClientWithOpts
-	_docker, err := docker.NewClientWithOpts(docker.FromEnv)
+	_docker, err := docker.NewClientWithOpts(docker.FromEnv, docker.WithHTTPClient(retryClient.StandardClient()))
```

Comment:
Per the comment above, the `defaultRetryWaitMax` does feel kinda long IMO.

I'm just thinking about like what if the `ContainerCreate` call succeeds but I wait 30 seconds to bomb out for the `ContainerStart` I feel like in those sequential situations it's likely going to bomb out anyway so waiting 30 seconds feels a bit overboard. Thoughts?

Reply:
I'm also thinking about reverting this change since it likely has unintended consequences like what you mentioned. My initial thought for this was specifically making [ImagePull](https://github.com/go-vela/worker/blob/19081a02a335086f774e8a73f595e9efb873659d/runtime/docker/image.go#L42) fail less often when connecting to somewhat unreliable repositories - suppose I'll consider adding some specific retry logic to that particular function in another pull request.

Reply:
Yeah, honestly, I think this package would have a ton of value in the sdk, but even then I would want to tune things because I don't think the `defaultRetryWaitMax` should be longer than like 10 seconds if its been added as a default to the entire client.